src/main/java/user_use_cases/UserGateway.java

```diff
@@ -0,0 +1,254 @@
+                        return_banned = true;
+                    }
+
+                    User return_user = new User(user[0], user[1], return_past_review, return_liked_reviews,
+                            Integer.parseInt(user[4]), return_banned);
+
+                    return return_user;
+                }
+            }
+        }catch(Exception e){
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+            return null;
+        }
+        return null;
+    }
+
+    @Override
+    public Boolean userExists(String username) {
+        try {
+            File file = new File(NAME_OF_USER_DATABASE);
+            Scanner scanner = new Scanner(file);
+            String delimiter = ",";
+            String line;
+
+            while(scanner.hasNext()){
+                line = scanner.nextLine();
+                String[] user = line.split(delimiter);
+
+                if (user[0].equals(username)){
+                    return true;
+                }
+            }
+            return false;
+        }catch(Exception e){
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Override
+    public String getPassword(String username) {
+        try {
+            File file = new File(NAME_OF_USER_DATABASE);
+            Scanner scanner = new Scanner(file);
+            String delimiter = ",";
+            String line;
+
+            while(scanner.hasNext()) {
+                line = scanner.nextLine();
+                String[] user = line.split(delimiter);
+
+                if (user[0].equals(username)) {
+                    return user[1];
+                }
+            }
+        }catch(Exception e){
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+            return null;
+        }
+        return null;
+    }
+
+    @Override
+    public void addUser(String username, String password) {
+        try {
+            File file = new File(NAME_OF_USER_DATABASE);
+            FileWriter writer = new FileWriter(file, true);
+            writer.append(username); // String username
+            writer.append(",");
+            writer.append(password); // String password
+            writer.append(",");
+            writer.append(""); // ArrayList<Review> past_reviews
+            writer.append(",");
+            writer.append(""); // ArrayList<String> likedReviews
+            writer.append(",");
+            writer.append("0"); // int received_reports = 0;
+            writer.append(",");
+            writer.append("0"); // boolean banned -> 0 - false, 1 - true
+            writer.append(",");
+            writer.append("\n");
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void banUser(String username) {
```

Comment:
I think you can remove banUser here; b/c Excalibur will decide whether one User should be banned and change the attributes, then ReportInteract will using UserGateway.updateUser to reflect the changes to the database